extensions/amp-moviads-player/0.1/amp-moviads-player.js

```diff
@@ -0,0 +1,713 @@
+    }
+  }
+
+  /** @override */
+  hideWhenEnd() {
+    if (this.config_.hideWhenEnd === true) {
+      this.element.remove();
+    }
+  }
+
+  /** @override */
+  hideWhenNoAd() {
+    if (this.config_.hideWhenNoAd === true) {
+      this.element.remove();
+    } else {
+      this.trackByPixel('movi_stat_start');
+    }
+  }
+
+  /** @override */
+  embedWatermark() {
+    if (this.config_.watermark !== null) {
+      const img = document.createElement('IMG');
+      img.src = this.config_.watermark;
+      img.className = 'amp-ima-player-watermark';
+      this.element.appendChild(img);
+    }
+  }
+
+  /** @override */
+  loadMethod() {
+    this.embedWatermark();
+    this.layoutLoad();
+    this.preconnectAds();
+  }
+
+  /** @override */
+  trackByPixel(event) {
+    const _this = this;
+    if (this.config_.stat.length) {
+      this.config_.stat.forEach(function(item) {
+        if (item.event === event && item.track.trim() != '') {
+          const img = document.createElement('IMG');
+          img.src = _this.replaceUrl(item.track);
+          img.className = 'pixel-none';
+          _this.element.appendChild(img);
+        }
+      });
+    }
+  }
+
+  /** @override
+   * @param {string} url
+   * @return {string}
+   * */
+  replaceUrl(url) {
+    return url
+      .replace(/__MID__/i, this.config_.mid)
+      .replace(/__LICENSE__/i, this.config_.license)
+      .replace(/__CONFIG__/i, this.config_.playerIdConf)
+      .replace(/__PID__/i, this.config_.pid);
+  }
+
+  /** @override */
+  LoadRecommendedVideo() {
+    const div = document.createElement('div');
+    div.className = 'amp-ima-player-recommended';
+    div.id = 'amp-ima-player-recommended';
+    if (this.config_.videoWall !== null) {
+      this.config_.videoWall.forEach(function(entry) {
+        const divContainer = document.createElement('div');
+        const ahref = document.createElement('a');
+        ahref.href = entry.uri;
+        const img = document.createElement('IMG');
+        img.src = entry.poster;
+        img.setAttribute('data-src', entry.uri);
+        ahref.appendChild(img);
+        const divTxt = document.createElement('div');
+        divTxt.className = 'title';
+        const ahrefDiv = document.createElement('a');
+        ahrefDiv.href = entry.uri;
+        ahrefDiv.appendChild(document.createTextNode(entry.title));
+        divTxt.appendChild(ahrefDiv);
+        divContainer.appendChild(ahref);
+        divContainer.appendChild(divTxt);
+        div.appendChild(divContainer);
+      });
+      this.element.appendChild(div);
+    }
+  }
+
+  /** @override */
+  functionAfterEnded() {
+    if (
+      this.config_.hideWhenEnd === false &&
+      typeof this.config_.functionAfterEnded !== 'undefined' &&
+      this.config_.functionAfterEnded !== null &&
+      this.config_.functionAfterEnded.length > 0 &&
+      typeof window[this.config_.functionAfterEnded] === 'function'
+    ) {
+      window[this.config_.functionAfterEnded]();
+    }
```

Comment:
You cannot execute arbitrary functions (e.g. dynamic name coming from an iframe) on the window document.